### ISSUE TITLE
Message dates not displayed correctly in topic headers

### ISSUE THREAD

**Issue by WesleyAC** [zulip/zulip-mobile#4756]
Topic headers show the date that the topic was created, rather than the date that the top visible message in the topic was sent. We should fix this to have the same behaviour as the webapp.

**Comment by saif-malik-01** [comment]
Hello, is this issue is still open? As I had made such apps in react native, I can solve this. Can you please assign me, so that I can start working on it. Thank you :)

**Comment by WesleyAC** [comment]
Hi! You're welcome to work on this, although it might be pretty tricky — this is in the [webview](https://github.com/zulip/zulip-mobile/blob/master/docs/background/webview.md) section of our code, which doesn't work in the same way as the React Native part of the code.

If you want to work on this, you're welcome to, although I'd also be happy to suggest an easier first issue if you'd prefer.

**Comment by saif-malik-01** [comment]
Yep , sure

**Comment by alya** [comment]
@hetanthakkar1  please post a comment describing your proposed approach when you're ready. I can assign the issue to you once you [have a rough plan](https://zulip.readthedocs.io/en/latest/overview/contributing.html#picking-an-issue-to-work-on). Thanks!

**Comment by Vatsalsoni13** [comment]
Hello @alya,
I tried to find where the bug was so the data in the stream is divided into 3 types: `time` , `header` and `message`. Currently, the header+date(which we see on screen) is provided by the `header` so the messages accompanied under the topic header are getting the date of `header`. My plan is since `time` data is always created when someone messages regardless it belongs to `header` or `message` so we can store the timestamp we get from `time` and fetch it in `header` and compare it (let's say timestamp from `time` is `x` and timestamp from `header` is y, we will perform 
```
if(x>y) { 
    use x on screen 
}
else {
    use y on screen
})
```

**Comment by chrisbobbe** [comment]
> Topic headers show the date that the topic was created, rather than the date that the top visible message in the topic was sent.

I don't think this is is true on `main`; could this have already been fixed somehow? Here's a screenshot with a [message from today](https://chat.zulip.org/#narrow/stream/48-mobile/topic/zulip-mobile.20.2F.20main/near/1346175):

<img width=300 src="https://user-images.githubusercontent.com/22248748/158655644-d20f6eec-c721-4df5-80af-878ee5d50c60.png" />

The "zulip-mobile / main" topic was created on 2021-09-08 with [this message](https://chat.zulip.org/#narrow/stream/48-mobile/topic/zulip-mobile.20.2F.20main/near/1253678); it wasn't created today.

The web app also appropriately says "today" for the header above the same message:

<img width="887" alt="image" src="https://user-images.githubusercontent.com/22248748/158655939-3fc4aa6f-5230-4c63-a418-e855dae1ef71.png">

**Comment by Vatsalsoni13** [comment]
Hello @chrisbobbe,
I think the issue is this 
WebApp:
![screen-capture](https://user-images.githubusercontent.com/53573748/158728199-78799c02-6302-458d-bebe-dda90c0f32ef.gif)
For eg: Under the topic 'Jumpy Toggle...', the topic was created on 12th March which shows correctly with that as you can see Alya's message on the 14th March which is updated in the header when it's on the top.
This behaviour is missing in the mobile app:

https://user-images.githubusercontent.com/53573748/158730107-6679e8e9-1591-483b-bdb6-e3544afd425f.mp4

When the 14th March message is on top the date in the header is not updated.

If you agree to this could you please have a look at my [above comment](https://github.com/zulip/zulip-mobile/issues/4756#issuecomment-1066825724)
Thanks :)
